include/proj/io.hpp

```diff
@@ -787,14 +807,77 @@ class PROJ_GCC_DLL PROJStringParser {
 
 // ---------------------------------------------------------------------------
 
-/** \brief Database context.
+/** \brief Database context interface.
  *
  * A database context should be used only by one thread at a time.
  */
-class PROJ_GCC_DLL DatabaseContext {
+class PROJ_GCC_DLL DatabaseContext :
```

Comment:
I'd have expected this interface to be at the SQL level. Here you have a few "high level" methods in the interface